Realm/RLMDictionary.h

```diff
@@ -0,0 +1,442 @@
+ */
+- (void)removeObjectForKey:(RLMKeyType)key;
+
+/**
+ Adds a given key-value pair to the dictionary.
+ */
+- (void)setObject:(nullable RLMObjectType)obj forKeyedSubscript:(RLMKeyType)key;
+
+/**
+ Adds a given key-value pair to the dictionary.
+ */
+- (void)setObject:(nullable RLMObjectType)anObject forKey:(RLMKeyType)aKey;
+
+/**
+  Adds to the receiving dictionary the entries from another dictionary.
+ 
+  @warning This method may only be called during a write transaction.
+
+  @param otherDictionary An enumerable object such as `NSDictionary` or `RLMDictionary` which contains objects of the
+         same type as the receiving dictionary.
+ */
+- (void)addEntriesFromDictionary:(id <NSFastEnumeration>)otherDictionary;
+
+#pragma mark - Querying a Dictionary
+
+/**
+ Returns all the objects matching the given predicate in the dictionary.
+
+ @param predicateFormat A predicate format string, optionally followed by a variable number of arguments.
+
+ @return                An `RLMResults` of objects that match the given predicate.
+ */
+- (RLMResults<RLMObjectType> *)objectsWhere:(NSString *)predicateFormat, ...;
+
+/// :nodoc:
+- (RLMResults<RLMObjectType> *)objectsWhere:(NSString *)predicateFormat args:(va_list)args;
+
+/**
+ Returns all the objects matching the given predicate in the dictionary.
+
+ @param predicate   The predicate with which to filter the objects.
+
+ @return            An `RLMResults` of objects that match the given predicate
+ */
+- (RLMResults<RLMObjectType> *)objectsWithPredicate:(NSPredicate *)predicate;
+
+/**
+ Returns a sorted `RLMResults` from the dictionary.
+
+ @param keyPath     The key path to sort by.
+ @param ascending   The direction to sort in.
+
+ @return    An `RLMResults` sorted by the specified key path.
+ */- (RLMResults<RLMObjectType> *)sortedResultsUsingKeyPath:(NSString *)keyPath ascending:(BOOL)ascending;
+
+/**
+ Returns a sorted `RLMResults` from the dictionary.
+
+ @param properties  An array of `RLMSortDescriptor`s to sort by.
+
+ @return    An `RLMResults` sorted by the specified properties.
+ */
+- (RLMResults<RLMObjectType> *)sortedResultsUsingDescriptors:(NSArray<RLMSortDescriptor *> *)properties;
+
+/**
+ Returns a distinct `RLMResults` from the dictionary.
+
+ @param keyPaths     The key paths to distinct on.
+
+ @return    An `RLMResults` with the distinct values of the keypath(s).
+ */
+- (RLMResults<RLMObjectType> *)distinctResultsUsingKeyPaths:(NSArray<NSString *> *)keyPaths;
+
+#pragma mark - Aggregating Property Values
+
+/**
+ Returns the minimum (lowest) value of the given property among all the values in the dictionary.
+
+     NSNumber *min = [object.dictionaryProperty minOfProperty:@"age"];
+
+ @warning You cannot use this method on `RLMObject`, `RLMDictionary`, and `NSData` properties.
+
+ @param property The property whose minimum value is desired. Only properties of
+                 types `int`, `float`, `double`, and `NSDate` are supported.
```

Comment:
This doc was taken from RLMArray, does this imply RLMArray is wrong too? Also it seems like Decimal128 is the only supported one left out. I don't really get stating 'You cannot use this method on' I think it would be better just to tell people what they can use it on?

Reply:
There's probably a whole pile of stale lists of types scattered around. Explicitly listing the types it works on and having it be implied that everything else doesn't work is probably the way to go. I think NSData was originally called out specifically in some places because there were some operations you'd expect to work that didn't but that was then fixed.